### PR TITLE
Core: Make sure StorybookError message shows up in browser console and interactions panel

### PR DESCRIPTION
**Pull request by kasperpeulen**
## What I did

Makes sure that the message is present in the stack of the built-in Error class in a way that works in all browsers.
Previously, the error structure was working in Safari and Firefox, but for some reason Chromium based browsers didn't have the error message in the stack.

This PR does the following:
- Updates the error class to ensure the message is contained in the stack
- Migrates all the errors to the new format
- Updates the local eslint plugin rules to conform to the new format
- Fixes the eslint config which was not executing some local eslint rules
- Updates necessary docs

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/core/src/server-errors.ts [hunk 1]
```diff
@@ -51,150 +51,131 @@ export enum Category {
...
-  readonly code = 1;
-
-  public readonly documentation = 'https://nx.dev/packages/storybook';
-
-  template() {
-    return dedent`
-      We have detected Nx in your project. Nx has its own Storybook initializer, so please use it instead.
-      Run "nx g @nx/storybook:configuration" to add Storybook to your project.
-    `;
+  constructor() {
```
Comment (yannbf) [inline code/core/src/server-errors.ts:54]: The diff is big I know, but make sure to review this file

File: code/core/src/preview-errors.ts [hunk 2]
```diff
@@ -33,361 +33,309 @@ export enum Category {
...
-  template() {
-    return dedent`
-    Couldn't find story matching id '${this.data.storyId}' after HMR.
-    - Did you just rename a story?
-    - Did you remove it from your CSF file?
-    - Are you sure a story with the id '${this.data.storyId}' exists?
-    - Please check the values in the stories field of your main.js config and see if they would match your CSF File.
-    - Also check the browser console and terminal for potential error messages.`;
+    super({
+      category: Category.PREVIEW_API,
```
Comment (yannbf) [inline code/core/src/preview-errors.ts:38]: The diff is big I know, but make sure to review this file

**Comment by yannbf** [comment]
I believe I made all the necessary changes, but it would be great to test this properly before merging.

File: code/core/src/preview-errors.ts [hunk 3]
```diff
@@ -33,361 +33,311 @@ export enum Category {
...
-    You have to install sharp in order to use image optimization features in Next.js.
-    `;
+  constructor(public data: { importType: string }) {
+    super({
+      category: Category.FRAMEWORK_NEXTJS,
+      code: 1,
+      message: dedent`
+        Tried to access sharp from "${data.importType}" but it was not available. You might be missing the required dependencies.
+      `,
+    });
```
Comment (kasperpeulen) [inline code/core/src/preview-errors.ts:310]: Seemed something went wrong here.

Comment (kasperpeulen) [reply code/core/src/preview-errors.ts]: I restored the original message

File: code/core/src/preview-errors.ts [hunk 4]
```diff
@@ -33,361 +33,314 @@ export enum Category {
...
-    return dedent`
-      Expected your framework's preset to export a \`renderToCanvas\` field.
-
-      Perhaps it needs to be upgraded for Storybook 6.4?`;
+  constructor() {
+    super({
+      category: Category.PREVIEW_API,
+      code: 4,
+      message: dedent`
+        Expected your framework's preset to export a \`renderToCanvas\` field.
```
Comment (shilman) [inline code/core/src/preview-errors.ts:91]: Shouldn't this be `renderer`?

File: code/core/src/preview-errors.ts [hunk 5]
```diff
@@ -33,361 +33,314 @@ export enum Category {
...
+
+        Perhaps it needs to be upgraded for Storybook 6.4?`,
```
Comment (shilman) [inline code/core/src/preview-errors.ts:93]:
```suggestion
        Perhaps it needs to be upgraded for Storybook 7.0?`,
```

File: code/core/src/preview-errors.ts [hunk 6]
```diff
@@ -33,361 +33,314 @@ export enum Category {
...
+        For example:
+
+        async play({ mount, canvasElement }) {
+          // 👈 mount should be called: await mount(); 
+          const canvas = within(canvasElement);
+          const button = await canvas.findByRole('button');
+          await userEvent.click(button);
+        };
+
+        Make sure to either remove it or call mount in your play function.
```
Comment (shilman) [inline code/core/src/preview-errors.ts:296]:
```suggestion
        Make sure to either remove the destructure or call mount in your play function.
```

File: code/core/src/storybook-error.ts [hunk 7]
```diff
@@ -22,20 +24,19 @@ export abstract class StorybookError extends Error {
 
   /**
    * Specifies the documentation for the error.
-   * - If `true`, links to a documentation page on the Storybook website (make sure it exists before enabling).
+   * - If `true`, links to a documentation page on the Storybook website (make sure it exists before enabling) – This is not implemented yet.
    * - If a string, uses the provided URL for documentation (external or FAQ links).
    * - If `false` (default), no documentation link is added.
    */
-  public documentation: boolean | string | string[] = false;
+  public documentation: boolean | string | string[];
```
Comment (shilman) [inline code/core/src/storybook-error.ts:31]: Should this also be `readonly` like the others?

**Review by shilman — APPROVED** [review]

File: code/core/src/storybook-error.ts (hunk 7 above)
Comment (kasperpeulen) [reply code/core/src/storybook-error.ts:31]: done

**Pull request merged**